src/main/java/org/testng/TestNG.java

```diff
@@ -717,17 +717,28 @@ public void addListener(ITestNGListener listener) {
       return;
     }
     if (listener instanceof ISuiteListener) {
-      m_suiteListeners.add((ISuiteListener) listener);
+      ISuiteListener suite = (ISuiteListener) listener;
+      if (! m_suiteListeners.containsKey(suite.getClass())) {
+        m_suiteListeners.put(suite.getClass(), (ISuiteListener) listener);
+      }
```

Comment:
From @missedone

> I think it'd be better to print some warning message that multi listener of the same type found, this might help people diagnose this behavior change?

Agree!